### PR TITLE
Implement sounds

### PR DESCRIPTION
**Pull request by simon816**
cherry-picked from #141 

Hopefully all the sound names are correct, no guarantees though :)

### CODE REVIEW

**Comment by me4502** [comment]
This looks good from a quick inspection.

**Comment by modwizcode** [comment]
This looks good, if someone else could go through the sound id's and make sure they're all correct then this can be merged.

**Comment by AIOOB** [comment]
I think you are currently missing all of these: http://pastebin.com/QQDcVt12, but the rest are all fine and named correctly.

**Comment by simon816** [comment]
@AIOOB Those sounds don't exist in the API, see [SoundTypes](https://github.com/SpongePowered/SpongeAPI/blob/master/src/main/java/org/spongepowered/api/effect/sound/SoundTypes.java).
Make a PR to the API if you think sponge should have those sounds :P

File: src/main/java/org/spongepowered/mod/registry/SpongeGameRegistry.java [hunk 1]
```diff
@@ -1095,6 +1098,226 @@ private void setGameModes() {
...
+        soundMappings.put("WITHER_IDLE", "mob.wither.idle");
+        soundMappings.put("WITHER_SHOOT", "mob.wither.shoot");
+        soundMappings.put("WITHER_SPAWN", "mob.wither.spawn");
+        soundMappings.put("WOLF_BARK", "mob.wolf.bark");
+        soundMappings.put("WOLF_DEATH", "mob.wolf.death");
+        soundMappings.put("WOLF_GROWL", "mob.wolf.growl");
+        soundMappings.put("WOLF_HOWL", "mob.wolf.howl");
+        soundMappings.put("WOLF_HURT", "mob.wolf.hurt");
+        soundMappings.put("WOLF_PANT", "mob.wolf.panting");
+        soundMappings.put("WOLF_SHAKE", "mob.wolf.panting");
```
Comment (AIOOB) [inline src/main/java/org/spongepowered/mod/registry/SpongeGameRegistry.java]: This should be mob.wolf.shake

File: src/main/java/org/spongepowered/mod/registry/SpongeGameRegistry.java [hunk 2]
```diff
@@ -1095,6 +1098,226 @@ private void setGameModes() {
...
+        soundMappings.put("PIG_WALK", "mob.pig.step");
+        soundMappings.put("RABBIT_IDLE", "mob.rabbit.idle");
+        soundMappings.put("RABBIT_HURT", "mob.rabbit.hurt");
+        soundMappings.put("RABBIT_DEATH", "mob.rabbit.death");
+        soundMappings.put("SHEEP_IDLE", "mob.sheep.say");
+        soundMappings.put("SHEEP_SHEAR", "mob.sheep.shear");
+        soundMappings.put("SHEEP_WALK", "mob.sheep.step");
+        soundMappings.put("SILVERFISH_HIT", "mob.silverfish.hit");
+        soundMappings.put("SILVERFISH_KILL", "mob.silverfish.kill");
+        soundMappings.put("SILVERFISH_IDLE", "mob.sheep.say");
```
Comment (AIOOB) [inline src/main/java/org/spongepowered/mod/registry/SpongeGameRegistry.java]: This should be mob.silverfish.say

**Pull request closed without merging**